exchanges/okx/okx.go

```diff
@@ -1060,149 +739,201 @@ func (ok *Okx) GetOneClickRepayHistory(ctx context.Context, after, before time.T
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getOneClickRepayHistoryEPL, http.MethodGet, common.EncodeURLValues("trade/one-click-repay-history", params), nil, &resp, request.AuthenticatedRequest)
+}
+
+// MassCancelOrder cancel all the MMP pending orders of an instrument family.
+// Only applicable to Option in Portfolio Margin mode, and MMP privilege is required
+func (ok *Okx) MassCancelOrder(ctx context.Context, instrumentType, instrumentFamily string, lockInterval int64) (*CancelMMPResponse, error) {
+	if instrumentType == "" {
+		return nil, fmt.Errorf("%w, empty instrument type", errInvalidInstrumentType)
+	}
+	if instrumentFamily == "" {
+		return nil, errInstrumentFamilyRequired
+	}
+	if lockInterval < 0 || lockInterval > 10000 {
+		return nil, fmt.Errorf("%w, lockInterval value range should be between 0 and 10000", errMissingIntervalValue)
+	}
+	arg := &struct {
+		InstrumentType   string `json:"instType,omitempty"`
+		InstrumentFamily string `json:"instFamily,omitempty"`
+		LockInterval     int64  `json:"lockInterval,omitempty"`
+	}{
+		InstrumentType:   instrumentType,
+		InstrumentFamily: instrumentFamily,
+		LockInterval:     lockInterval,
+	}
+	var resp *CancelMMPResponse
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, tradeOneClickRepayEPL, http.MethodPost, "trade/mass-cancel", arg, &resp, request.AuthenticatedRequest)
+}
+
+// CancelAllMMPOrdersAfterCountdown cancel all MMP pending orders after the countdown timeout.
+// Only applicable to Option in Portfolio Margin mode, and MMP privilege is required
+func (ok *Okx) CancelAllMMPOrdersAfterCountdown(ctx context.Context, timeout int64, orderTag string) (*CancelResponse, error) {
+	if (timeout != 0) && (timeout < 10 || timeout > 120) {
+		return nil, fmt.Errorf("%w, Range of value can be 0, [10, 120]", errCountdownTimeoutRequired)
+	}
+	arg := &struct {
+		TimeOut  int64  `json:"timeOut,string"`
+		OrderTag string `json:"tag,omitempty"`
+	}{
+		TimeOut:  timeout,
+		OrderTag: orderTag,
+	}
+	var resp *CancelResponse
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, cancelAllAfterCountdownEPL, http.MethodPost, "trade/cancel-all-after", arg, &resp, request.AuthenticatedRequest)
+}
+
+// GetTradeAccountRateLimit get account rate limit related information.
+// Only new order requests and amendment order requests will be counted towards this limit. For batch order requests consisting of multiple orders, each order will be counted individually
+func (ok *Okx) GetTradeAccountRateLimit(ctx context.Context) (*AccountRateLimit, error) {
+	var resp *AccountRateLimit
+	return resp, ok.SendHTTPRequest(ctx, exchange.RestSpot, getTradeAccountRateLimitEPL, http.MethodGet, "trade/account-rate-limit", nil, &resp, request.AuthenticatedRequest)
+}
+
+// OrderPreCheck precheck the account information before and after placing the order.
```

Comment:
* precheck => prechecks
* But actually outside the context of func naming, this is just "checks" or "returns"
* Switched "the order" to "a potential" to make it clear it's not actually placing the order
* I'd also suggest that to match `Place*` funcs, we name this `PreCheckOrder`.
```suggestion
// OrderPreCheck returns the account information before and after placing a potential order
```